payment-component/vanilla-js/apps/client/scripts/checkout.js

```diff
@@ -0,0 +1,82 @@
+async function getData() {
+    const url = "http://localhost:8082/payment-intent";
+    try {
+        const response = await fetch(url, {
+            method: "POST",
+        });
+        if (!response.ok) {
+            throw new Error(`Response status: ${response.status}`);
+        }
+        const json = await response.json();
+        console.log(json.id);
+        const chargebee = window.Chargebee.init({
+            site: env.site,
+            publishableKey: env.publishableKey,
+        });
+
+        const componentOptions = {
+            locale: "fr",
+            style: {
+                theme: {
+                    accentColor: "gold",
+                    appearance: "light"
+                },
+                variables: {
+                    spacing: 2,
+                    accentIndicator: "#ffff00",
+                }
+            }
+        };
+
+        const components = chargebee.components(componentOptions);
+
+        const onSuccess = (payment_intent) => {
+            console.log(payment_intent);
+        }
+
+        const onError = (error) => {
+            // handle payment errors here
+            console.log(error);
+        }
+
+        const onPaymentMethodChange = (error) => {
```

Comment:
@cb-amalgeorge is this called whenever the customer selects a payment method on the UI? What does the `error` contain in such a case?